### PR TITLE
Artefact page improvements

### PR DESCRIPTION
**Pull request by omar-selo**
## Description

A bunch of improvements to artefact page:
- Remove top right legends and expand the test executions list to take up the space above
- Remove artefact build expandables this means that now the list of test execution expandables will build and destroy widgets as you scroll to improve efficiency
- Make expanding a test execution automatically expand failed tests (if there are any)
- Make expanding a test execution automatically expand test events (if it didn't complete)
- Prefetch test results so that expanding a test execution opens them immediately
- Remove the animation of opening and closing an expandable to make the experience snappier
- Replace horizontal scrolling of test events with tooltips

## Resolved issues

Resolves https://warthogs.atlassian.net/browse/RTW-335

## Documentation

None

## Web service API changes

None

## Tests

See video below for the overall experience:
[Screencast from 2024-07-30 14-05-18.webm](https://github.com/user-attachments/assets/9b6b8bbe-53be-4e23-bd68-1fbd93395705)

Image below for test event tooltip:
Screenshot from 2024-07-30 14-08-09

### CODE REVIEW

**Comment by andrejvelichkovski** [comment]
Since that now we don't have a legend, do you think we should add tooltips to the environment icons here:

Screenshot from 2024-07-31 10-10-46

File: frontend/lib/ui/artefact_page/artefact_page_side.dart [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+                  loading: () => const YaruCircularProgressIndicator(),
+                  error: (e, stack) =>
+                      Center(child: Text('Error:\n$e\nStackTrace:\n$stack')),
+                  data: (artefactBuilds) => const PageFiltersView(
+                    searchHint: 'Search by environment name',
+                    width: double.infinity,
+                  ),
+                );
+              },
+            ),
```
Comment (andrejvelichkovski) [inline frontend/lib/ui/artefact_page/artefact_page_side.dart:40]: If I'm not wrong this code is rendering the side filter, do you think we can move it to a separate widget, perhaps a private one?

File: frontend/lib/ui/artefact_page/test_event_log_expandable.dart [hunk 2]
```diff
@@ -18,54 +18,57 @@ class TestEventLogExpandable extends ConsumerWidget {
...
+                  style: TextStyle(fontStyle: FontStyle.italic),
                 ),
               ),
-              DataColumn(
-                label: Expanded(
-                  child: Text(
-                    'Timestamp',
-                    style: TextStyle(fontStyle: FontStyle.italic),
-                  ),
+            ),
```
Comment (andrejvelichkovski) [inline frontend/lib/ui/artefact_page/test_event_log_expandable.dart:34]: It seems that here we have some formatting changes, should we rebase to main to use the formatter you added in the previous PR?

**Review by andrejvelichkovski — APPROVED** [review]
Took me quite a while to go through all the changes, but I couldn't find anything that would break the main functionalities, as most of the changes are moving code around to optimize it. I really hope this will improve the review experience.

File: frontend/lib/ui/artefact_page/test_event_log_expandable.dart (hunk 2 above)
Comment (omar-selo) [reply frontend/lib/ui/artefact_page/test_event_log_expandable.dart:34]: There's no formatting changes here (already merged with main). But I've removed the `SingleChildScrollView`

**Review by andrejvelichkovski — APPROVED** [review]
LGTM!

**Pull request merged**